### PR TITLE
Add some output fields to RRFS

### LINKED ISSUE #770: Add instantaneous direct and diffuse irradiance output for RRFS
We need to enable output of instantaneous direct and diffuse irradiance in RRFS, since they were output in HRRR.

### LINKED ISSUE #773: Enable output of land surface model fields (LAI, SOILM, WILT) in RRFS
These LSM fields are output in predecessor models, so we will need them in RRFS.

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
This PR addresses issue #770 and #773.

Add direct and diffuse solar irradiance to RRFS output, as well as some LSM related output that we need to produce (to match HRRR).  

The new code was tested on Jet for RRFS_NA_3km system.

### CODE REVIEW

**Comment by FernandoAndrade-NOAA** [comment]
Hi @EricJames-NOAA, please be sure to create/link a git issue and add the appropriate entries in the changelogs within the sorc/ files updated, thank you!

**Comment by EricJames-NOAA** [comment]
> Hi @EricJames-NOAA, please be sure to create/link a git issue and add the appropriate entries in the changelogs within the sorc/ files updated, thank you!

@FernandoAndrade-NOAA thanks.  I can't quite figure out how to link my github issue #773 to this PR....maybe I don't have permission to do that?

**Comment by FernandoAndrade-NOAA** [comment]
> > Hi @EricJames-NOAA, please be sure to create/link a git issue and add the appropriate entries in the changelogs within the sorc/ files updated, thank you!
> 
> @FernandoAndrade-NOAA thanks. I can't quite figure out how to link my github issue #773 to this PR....maybe I don't have permission to do that?

It would just be a section on the right labeled "Development" that should automatically close the issue upon PR merge, but you may be right it might be a permissions issue. I've gone ahead and added it manually thanks! For any future PRs, it's fine to just add a link within the PR description, thanks!

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Please sync your branch with the latest UPP develop branch and provide me the sample RRFS model output in netcdf for my testing.

File: parm/fv3lam_rrfs.xml [hunk 1]
```diff
@@ -399,6 +399,12 @@
        <scale>4.0</scale>
        </param>
 
+       <param>
+       <shortname>SOILM_ON_DEPTH_BEL_LAND_SFC</shortname>
+       <pname>SOILM</pname>
```
Comment (WenMeng-NOAA) [inline parm/fv3lam_rrfs.xml]: @EricJames-NOAA Please change SOILM into SOILMOI. The grib2 name 'SOILM' was deprecated in the grib2 tables.

File: parm/fv3lam_rrfs.xml [hunk 2]
```diff
@@ -2419,6 +2439,12 @@
        <scale>-2.0</scale>
        </param>
 
+       <param>
+       <shortname>SOILM_ON_DEPTH_BEL_LAND_SFC</shortname>
+       <pname>SOILM</pname>
```
Comment (WenMeng-NOAA) [inline parm/fv3lam_rrfs.xml]: @EricJames-NOAA Please change SOILM into SOILMOI. The grib2 name 'SOILM' was deprecated in the grib2 tables.

Comment (EricJames-NOAA) [reply parm/fv3lam_rrfs.xml]: @WenMeng-NOAA welcome back!  I will work on finalizing the PR this week.

**Comment by EricJames-NOAA** [comment]
@WenMeng-NOAA we have some sample history files (9-h forecast) here on Hera:

/scratch1/BMC/wrfruc/mhu/temp/

Thanks!

File: parm/post_avblflds.xml [hunk 3]
```diff
@@ -6931,7 +6931,7 @@
        <post_avblfldidx>755</post_avblfldidx>
        <shortname>HWP_ON_SURFACE</shortname>
        <longname>Hourly Wildfire Potential on surface</longname>
-       <pname>FWINX</pname>
+       <pname>WFIREPOT</pname>
```
Comment (WenMeng-NOAA) [inline parm/post_avblflds.xml:6934]: @EricJames-NOAA May I know any reason for a new grib2 name 'WFIREPOT' replacing 'FWINX' for RRFS products?

Comment (EricJames-NOAA) [reply parm/post_avblflds.xml:6934]: @WenMeng-NOAA I worked with @AndrewBenjamin-NOAA to define a new GRIB2 name for this field.  It is a new diagnostic product that we have developed at NOAA GSL, on which we are about to submit a paper to Weather and Forecasting.  I can provide more details if you are interested.

Comment (WenMeng-NOAA) [reply parm/post_avblflds.xml:6934]: @EricJames-NOAA In the grib2 table [4.2-2-4](https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-2-4.shtml), FWINDX is defined as fire wealth index(Canadian Forest Service) , and WFIREPOT is defined as wildfire potential( defined by GSL). Do you know this wildfire product is currently generated in rap/hrrr/nam? If it is a new product for RRFS only, it is fine for me for using new grib2 name 'WFIREPOT'.

Comment (EricJames-NOAA) [reply parm/post_avblflds.xml:6934]: @WenMeng-NOAA this is a new product only available for RRFS; it is not in RAP/HRRR/NAM.  Sounds good, thanks!

Comment (WenMeng-NOAA) [reply parm/post_avblflds.xml:6934]: @EricJames-NOAA Thanks for clarifying.

File: parm/params_grib2_tbl_new [hunk 4]
```diff
@@ -367,6 +367,7 @@
    0   1   121   0 FSNOWC
    0   6    36   0 FSTRPC
    2   4     5   0 FWINX
+   2   4    26   0 WFIREPOT
```
Comment (WenMeng-NOAA) [inline parm/params_grib2_tbl_new]: @EricJames-NOAA The file params_grib2_tbl_new should be sorted by grib2 names based on params_grib2_tbl_new.text. Please pick up my params_grib2_tbl_new from /scratch1/NCEPDEV/stmp2/Wen.Meng/eric_james/UPP and commit it to your branch.

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA My tests are at /scratch1/NCEPDEV/stmp2/Wen.Meng:
'''
post_rrfs_2024021416-before:  test with the current develop branch

post_rrfs_2024021416-after:  test with your branch
I verified all your proposed changes were generated in it.
```
Please verify my tests.

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Please sync your branch with the latest UPP develop branch again.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs were completed on WCOSS2 and Hera. The baseline recreations are needed for fv3r and 3drtma with this PR.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA You may start the UPP RTs on Hera and Orion. Please refer to my testing on Hera at /scratch1/NCEPDEV/stmp2/Wen.Meng. The datasets of NATLEV and PRSLEV in both fv3r and 3drtma have the following changes:
```
Records of NATLEV: 1570 -> 1648
Records of PRSLEV: 1078 -> 1090
```

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA My tests are at /scratch1/NCEPDEV/stmp2/Wen.Meng: ''' post_rrfs_2024021416-before: test with the current develop branch
> 
> post_rrfs_2024021416-after: test with your branch I verified all your proposed changes were generated in it.
> 
> ```
> Please verify my tests.
> ```

Your results look consistent with mine!  Thanks for your test!

**Comment by FernandoAndrade-NOAA** [comment]
> @FernandoAndrade-NOAA You may start the UPP RTs on Hera and Orion. Please refer to my testing on Hera at /scratch1/NCEPDEV/stmp2/Wen.Meng. The datasets of NATLEV and PRSLEV in both fv3r and 3drtma have the following changes:
> 
> ```
> Records of NATLEV: 1570 -> 1648
> Records of PRSLEV: 1078 -> 1090
> ```

Hera is still in queue. It likely won't finish until tomorrow. My Orion results seem to match with yours:
```
1648 records for NATLEV within fv3r and 3drtma
1103 records for PRSLEV within rtma
1090 records for PRSLEV within fv3r
```

Hercules is unfortunately running into improper file generation again. We'll track that in a separate issue. I'll update here with the Hera results tomorrow.

**Comment by FernandoAndrade-NOAA** [comment]
Hera has finished RTs as well with the expected record count changes within NATLEV and PRSLEV for fv3r and 3drtma. @WenMeng-NOAA FYI

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
@WenMeng-NOAA Are the changelog comment additions to INITPOST and SRFCE sufficient for ALLOCATE and DEALLOCATE or do we need an additional commit for those as well?

**Comment by WenMeng-NOAA** [comment]
> @WenMeng-NOAA Are the changelog comment additions to INITPOST and SRFCE sufficient for ALLOCATE and DEALLOCATE or do we need an additional commit for those as well?

@FernandoAndrade-NOAA The subroutines ALLOCATE and DEALLOCATE are for variable allocation and deallocation.  

> @WenMeng-NOAA Are the changelog comment additions to INITPOST and SRFCE sufficient for ALLOCATE and DEALLOCATE or do we need an additional commit for those as well?

The change logs in INITPOST_NETCDF.f and SURFCE.f would be sufficient.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**